### PR TITLE
Preliminary bhyve gdb stub support

### PR DESCRIPTION
**Pull request by markjdb**
These patches provide a very limited gdb stub for aarch64 guests. In particular:
- Breakpoints and single-stepping are not yet implemented. (Breakpoints do work but need single-stepping to be useful.)
- The stub uses a fixed register layout for aarch64 and thus cannot be used to fetch capability registers, so there is no ability to read registers from a purecap guest. The solution to this will be to implement support for gdb's XML register descriptions.
However, it's enough to help debug certain types of problems, so we might as well try to include it.

### CODE REVIEW

File: usr.sbin/bhyve/gdb.c [hunk 1]
```diff
@@ -187,7 +210,85 @@ static const int gdb_regsize[] = {
...
+	VM_REG_GUEST_X28,
+	VM_REG_GUEST_X29,
+	VM_REG_GUEST_LR,
+	VM_REG_GUEST_SP,
+	VM_REG_GUEST_PC,
+	VM_REG_GUEST_CPSR,
+};
+
+static const int gdb_regsize[] = {
+	8,
```
Comment (brooksdavis) [inline usr.sbin/bhyve/gdb.c:254]: Maybe consider adding designators `[0] = 8]` for readability? If you're willing to use a gnu extension (which clang supports) you can even make this super compact:
```
const int gdb_regsize[] = {
	[0 ... 32] = 8,
	4,
};
```

Comment (brooksdavis) [reply usr.sbin/bhyve/gdb.c:254]: I also wonder if there should be a _Static_assert that the two arrays are the same size?

**Review by brooksdavis — APPROVED** [review]
As far as I understand what's going on this seems good. A couple comments on an array initialization that could be clearer, but matches existing amd64 style so is fine.

File: usr.sbin/bhyve/gdb.c (hunk 1 above)
Comment (markjdb) [reply usr.sbin/bhyve/gdb.c:254]: I added the _Static_assert, but indeed will avoid changing the amd64 code for now. Those arrays will be mostly obsolete once support for xmlRegisters is done.

**Pull request merged**